hbase-client/src/main/java/org/apache/hadoop/hbase/security/provider/SaslClientAuthenticationProviders.java

```diff
@@ -148,8 +150,9 @@ static void addExplicitProviders(Configuration conf,
       // Instantiate it
       SaslClientAuthenticationProvider provider;
       try {
-        provider = (SaslClientAuthenticationProvider) clz.newInstance();
-      } catch (InstantiationException | IllegalAccessException e) {
+        provider = (SaslClientAuthenticationProvider) clz.getConstructor().newInstance();
```

Comment:
This modification causes TestSaslClientAuthenticationProviders#testDifferentConflictingImplementationsFail test failure. I'll look into it later.